source/server/admin/config_dump_handler.cc

```diff
@@ -129,11 +143,21 @@ Http::Code ConfigDumpHandler::handlerConfigDump(absl::string_view url,
   if (resource.has_value()) {
     auto err = addResourceToDump(dump, mask, resource.value(), include_eds);
     if (err.has_value()) {
+      response_headers.addReference(Http::Headers::get().XContentTypeOptions,
+                                    Http::Headers::get().XContentTypeOptionValues.Nosniff);
+      response_headers.setReferenceContentType(Http::Headers::get().ContentTypeValues.Text);
       response.add(err.value().second);
       return err.value().first;
     }
   } else {
-    addAllConfigToDump(dump, mask, include_eds);
+    auto err = addAllConfigToDump(dump, mask, include_eds);
+    if (err.has_value()) {
+      response_headers.addReference(Http::Headers::get().XContentTypeOptions,
+                                    Http::Headers::get().XContentTypeOptionValues.Nosniff);
+      response_headers.setReferenceContentType(Http::Headers::get().ContentTypeValues.Text);
+      response.add(err.value().second);
+      return err.value().first;
```

Comment:
5-line stanza repeated twice. Maybe try structuring like this to avoid:
```
if (!err.has_value()) {
  err = addAllConfigToDump(dump, mask, include_eds);
}
if (err.has_value()) {
  // 5-line error stanza
}
```
or maybe even
```
if (err.has_value() || (err = addAllConfigToDump(dump, mask, include_eds)).has_value()) {
```
or something like that? Not sure how much I like that second pattern :)